vmaas/reposcan/database/csaf_store.py

```diff
@@ -46,7 +47,132 @@ def _save_csaf_files(self, csaf_files: CsafFiles) -> list[int]:
         self.conn.commit()
         return db_ids
 
+    def _get_product_rows(self, products_by_status):
+        to_associate_ids = []
+        to_insert = defaultdict(list)
+        try:
+            cur = self.conn.cursor()
+            for status, products in products_by_status.items():
+                if not products:
+                    continue
+                rows = execute_values(cur, """SELECT id, cpe, package, module
```

Comment:
The select works, except for a case when module is NULL.

```
SELECT id, cpe, package, module from csaf_products where (cpe, package, module) in (('cpe:/o:redhat:enterprise_linux:8', 'dotnet8.0', NULL));
```
returns nothing, even though such row is in the table

something like this could select correctly but I haven't tested it very well and it could be written nicer
```
rows = cur.execute("""SELECT id, cpe, package, module
                                      FROM csaf_products
                                      WHERE ((cpe, package, module) in %s AND module IS NOT NULL)
                                         OR ((cpe, package) in %s AND module IS NULL)""",
                                   (tuple([row for row in products]),
                                    tuple((cpe, package) for cpe, package, _ in products)))
```

Reply:
Arguments have been simplified and I tested it against NULL modules and it worked.